### PR TITLE
Service Endpoint Policies Swagger

### PR DESCRIPTION
**Pull request by chandrasekarsrinivasan**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+             "serviceResources": [
+                "/subscriptions/subid1",
+                "/subscriptions/subid1/resourceGroups/storageRg",
+                "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+             ]
+            },
+           "provisioningState" : "Succeeded"
+           }
+          ]
+         },
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json]: Remove extra comma

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json]: remove extra comma

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "serviceEndpointPolicyDefinitionName": "testDefinition",
+    "properties": {
+     "description": "Storage Service EndpointPolicy Definition",
+     "service": "Microsoft.Storage",
+     "serviceResources": [
+        "/subscriptions/subid1",
+        "/subscriptions/subid1/resourceGroups/storageRg",
+        "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+     ]
+    },
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json]: Remove extra comma

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        "id":"/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/serviceEndpointPolicies/testPolicy/serviceEndpointPolicyDefinitions/testDefinition",
+        "properties": {
+         "description": "Storage Service EndpointPolicy Definition",
+         "service": "Microsoft.Storage",
+         "serviceResources": [
+            "/subscriptions/subid1",
+            "/subscriptions/subid1/resourceGroups/storageRg",
+            "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+         ]
+    },
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json]: comma

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        "id":"/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/networkSecurityGroups/testnsg/securityRules/rule1",
+        "properties": {
+         "description": "Storage Service EndpointPolicy Definition",
+         "service": "Microsoft.Storage",
+         "serviceResources": [
+            "/subscriptions/subid1",
+            "/subscriptions/subid1/resourceGroups/storageRg",
+            "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+         ]
+        },
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json]: comma

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionList.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            "id":"/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/serviceEndpointPolicies/testPolicy/serviceEndpointPolicyDefinitions/testDef",
+            "properties": {
+             "description": "Storage Service EndpointPolicy Definition",
+             "service": "Microsoft.Storage",
+             "serviceResources": [
+                "/subscriptions/subid1",
+                "/subscriptions/subid1/resourceGroups/storageRg",
+                "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+             ]
+            },
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionList.json]: comma

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json [hunk 7]
```diff
@@ -0,0 +1,634 @@
...
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/ServiceEndpointPolicies": {
+      "get": {
+        "tags": [
+          "ServiceEndpointPolicies"
+        ],
+        "operationId": "ServiceEndpointPolicies_ListAll",
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: This method should be called "List"

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json [hunk 8]
```diff
@@ -0,0 +1,634 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/serviceEndpointPolicies": {
+      "get": {
+        "tags": [
+          "serviceEndpointPolicies"
+        ],
+        "operationId": "serviceEndpointPolicies_List",
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: This should be called "ListByResourceGroup"

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json (hunk 8 above)
Comment (chandrasekarsrinivasan) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: Fixed

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json (hunk 7 above)
Comment (chandrasekarsrinivasan) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: Fixed

**Comment by jianghaolu** [comment]
@chandrasekarsrinivasan Can you push your fixes?

Also you need to sync with Network-August-Release branch?

**Comment by chandrasekarsrinivasan** [comment]
I pushed the fixes.  Also, yes, I need this on network-august-release

**Comment by jianghaolu** [comment]
@chandrasekarsrinivasan I meant that you have to pull from Network-August-Release.

**Comment by chandrasekarsrinivasan** [comment]
Yes, I already pulled. I pulled again now.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json [hunk 9]
```diff
@@ -0,0 +1,634 @@
...
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "ServiceEndpointPolicyName",
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: Should be "serviceEndpointPolicyName"

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json (hunk 9 above)
Comment (chandrasekarsrinivasan) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: done

Comment (chandrasekarsrinivasan) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: done

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json [hunk 10]
```diff
@@ -0,0 +1,634 @@
...
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create security rule": { "$ref": "./examples/ServiceEndpointPolicyDefinitionCreate.json" }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/serviceEndpointPolicies/{ServiceEndpointPolicyName}/serviceEndpointPolicyDefinitions": {
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: {ServiceEndpointPolicyName} should be {serviceEndpointPolicyName}

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json (hunk 10 above)
Comment (chandrasekarsrinivasan) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: done

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyGet.json [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+            "properties": {
+             "description": "Storage Service EndpointPolicy Definition",
+             "service": "Microsoft.Storage",
+             "serviceResources": [
+                "/subscriptions/subid1",
+                "/subscriptions/subid1/resourceGroups/storageRg",
+                "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+             ]
+            },
+           "provisioningState" : "Succeeded"
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyGet.json]: `provisioningState` is not defined in the `ServiceEndpointPolicyDefinition` definition

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json [hunk 12]
```diff
@@ -0,0 +1,63 @@
+{
+  "parameters" : {
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json]: Missing parameter `serviceEndpointPolicyName`

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+            "properties": {
+             "description": "Storage Service EndpointPolicy Definition",
+             "service": "Microsoft.Storage",
+             "serviceResources": [
+                "/subscriptions/subid1",
+                "/subscriptions/subid1/resourceGroups/storageRg",
+                "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+             ]
+            },
+           "provisioningState" : "Succeeded"
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json]: Same

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json]: Same

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyListAll.json [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+                "properties": {
+                 "description": "Storage Service EndpointPolicy Definition",
+                 "service": "Microsoft.Storage",
+                 "serviceResources": [
+                    "/subscriptions/subid1",
+                    "/subscriptions/subid1/resourceGroups/storageRg",
+                    "/subscriptions/subid1/resourceGroups/storageRg/providers/Microsoft.Storage/storageAccounts/stAccount"
+                 ]
+                },
+               "provisioningState" : "Succeeded"
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyListAll.json]: Same

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyListAll.json [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyListAll.json]: Same - you probably missed defining it in your spec?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionGet.json [hunk 17]
```diff
@@ -0,0 +1,28 @@
+{
+  "parameters" : {
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionGet.json]: Missing parameter `serviceEndpointPolicyName`

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionGet.json [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+        "properties": {
+          "provisioningState": "Succeeded",
+          "protocol": "*",
+          "sourcePortRange": "*",
+          "destinationPortRange": "80",
+          "sourceAddressPrefix": "*",
+          "destinationAddressPrefix": "*",
+          "access": "Allow",
+          "priority": 130,
+          "direction": "Inbound"
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionGet.json]: Additional properties not allowed:
    direction,priority,access,destinationAddressPrefix,sourceAddressPrefix,destinationPortRange,sourcePortRange,protocol are not defined in `ServiceEndpointPolicyDefinition`

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json [hunk 19]
```diff
@@ -0,0 +1,50 @@
+{
+  "parameters" : {
+    "api-version": "2018-07-01",
+    "subscriptionId" : "subid",
+    "resourceGroupName" : "rg1",
+    "serviceEndpointPolicyName" : "testPolicy",
+    "serviceEndpointPolicyDefinitionName": "testDefinition",
+    "properties": {
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionCreate.json]: wrap this in `ServiceEndpointPolicyDefinitionParameters`

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json [hunk 20]
```diff
@@ -0,0 +1,634 @@
...
+          },
+          {
+            "name": "serviceEndpointPolicyDefinitionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the service endpoint policy definition name."
+          },
+          {
+            "name": "ServiceEndpointPolicyDefinitionParameters",
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/serviceEndpointPolicy.json]: BTW parameter names should be camel cased. I missed this earlier my bad

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json (hunk 12 above)
Comment (chandrasekarsrinivasan) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyCreate.json]: done.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionGet.json (hunk 17 above)
Comment (chandrasekarsrinivasan) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/ServiceEndpointPolicyDefinitionGet.json]: done

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**